codes/net/dragonfly-dally.h

```diff
@@ -37,27 +39,38 @@ struct terminal_dally_message
   tw_lpid sender_mn_lp; // source modelnet id
  /* destination terminal ID of the dragonfly */
   tw_lpid dest_terminal_lpid;
-  int dfdally_dest_terminal_id; //this is the terminal id in the dfdally network in range [0-total_num_terminals)
+  unsigned int dfdally_src_terminal_id;
+  unsigned int dfdally_dest_terminal_id; //this is the terminal id in the dfdally network in range [0-total_num_terminals)
   /* source terminal ID of the dragonfly */
   unsigned int src_terminal_id;
   /* message originating router id. MM: Can we calculate it through
    * sender_mn_lp??*/
   unsigned int origin_router_id;
 
+  int app_id; //id of the job associated with this terminal TODO - this will cause a problem if multiple job workload LPs are mapped to one terminal
+
   /* number of hops traversed by the packet */
   short my_N_hop;
   short my_l_hop, my_g_hop;
+  short my_hops_cur_group;
   short saved_channel;
   short saved_vc;
 
   int next_stop;
 
+  //encoded time when received at a router
+  tw_stime this_router_arrival;
+  //encoded time when departed from router
+  tw_stime this_router_ptp_latency;
+  short ptp_latency_set;
```

Comment:
Make a todo here - ptp_latency_set handles the case where a R_SEND event doesn't actually result in a packet being sent (full vc etc). This could maybe be replaced by just looking at a bitfield value and save a couple bytes.

Reply:
Actually this was already done, removing this field.